### PR TITLE
 (PUP-6702) Correctly manage paused Windows services

### PR DESCRIPTION
**Pull request by ekinanp**

### CODE REVIEW

**Comment by ekinanp** [comment]
jenkins please test this on windows10ent-64a,windows10ent-32a,windows2008r2-64a,windows2012r2-64a,windows2012r2_ja-64a%7Blocale=ja%7D,windows2016-64a

File: lib/puppet/provider/service/windows.rb [hunk 1]
```diff
@@ -56,6 +56,13 @@ def enabled?
   end
 
   def start
+    if status == :paused
+      Puppet::Util::Windows::Service.resume(@resource[:name])
```
Comment (ekinanp) [inline lib/puppet/provider/service/windows.rb]: I double-checked that paused and disabled services can be resumed, so it is safe to place this code here.

File: lib/puppet/util/windows/service.rb [hunk 2]
```diff
@@ -331,6 +331,30 @@ def stop(service_name)
     end
     module_function :stop
 
+    # Resume a paused windows service
+    #
+    # @param [:string] service_name name of the service to resume
```
Comment (caseywilliams) [inline lib/puppet/util/windows/service.rb]: I believe the correct rdoc format is `@param [String] service_name`

Comment (ekinanp) [reply lib/puppet/util/windows/service.rb]: Thanks for pointing that out. It looks like we may need a separate PR to update the rdoc comments b/c there's some other formatting errors in the module (e.g. [here](https://github.com/puppetlabs/puppet/blob/master/lib/puppet/util/windows/service.rb#L258))

**Pull request merged**